openedx/openedx-platform
# feat: support static reference in LibraryContentBlock

Requires:

* https://github.com/openedx/edx-platform/pull/33263

## Description

This PR adds the ability for course authors to reference specific pieces of content from a content library in their course using the library xblock. It does so for v1 and v2 xblocks.

It does so by implementing the changes described in the [ADR which describes the new desired schema for the block](https://github.com/openedx/edx-platform/pull/33231)

## Testing instructions

To test this PR:

1. Create a v1 and a v2 content library, "v1 lib" and "v2 lib." Make sure each has some problems in it.
2. In a unit in a course, create 2 library blocks. associate v1 lib with the first block, and v2 lib with the second.
  
5. Test Randomized content behavior: with shuffle = true and max_count = 1, save and publish the blocks.
expected: you should see 1 block at random per user per block. Try again with max_count = 2, and then max_count =  -1. You should then see 2 and all blocks respectively. 

7. Test Specific content reference behavior. With shuffle = false and manual = true, select one piece of content from the v1 and v2 libs. Republish. This content should then be displayed to the user.
